ejb/WFLY-6143-Ability-to-add-a-server-side-configured-Interceptor-for-the-applications-without-changing-the-deployments.adoc

```diff
@@ -0,0 +1,129 @@
+= Ability to add a server-side configured Interceptor for the applications without changing the deployments
+:author:            Tomasz Adamski
+:email:             tomasz.adamski@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+The goal of this issue is to implement server side EJB interceptors that can be configured system-wide through ejb3 subsystem configuration.
+
+=== Issue
+* https://issues.jboss.org/browse/WFLY-6143
+
+=== Dev Contacts
+
+* mailto:tadamski@redhat.com
+
+=== QE Contacts
+
+=== Testing By
+[ ] Engineering
+
+[x] QE
+
+=== Affected Projects or Components
+EJB
+
+=== WildFLY custom interceptors
+
+WildFly allows users to implement their own intereceptors as a part of the deployment (https://docs.jboss.org/author/display/WFLY10/Container+interceptors). Most important characteristics of those are as follows:
+
+* interceptors are POJO classes marked whose interceptor methods are marked by annotations
+* interceptor provider can specify in the descriptor the rules to match interceptors with specific beans
+
+In order to keep programming model consistent server-side interceptors would use the same APIs (POJO with javax.interceptor annotation's). Contrarily to customer interceptors, server-interceptors must be decoupled from the deployments. As a result, server-interceptors would not provide the ability to match them with specific beans.
+
+
+== Requirements
+* Interceptors are implemented and packed inside the custom modules
+* Interceptors should be configured in the subsystem configuration and not in deployments
+
+== Nice to have requirements
+* There should be a number of binding points in which the interceptors should be executed (f.e. after security)
```

Comment:
Per the "Binding points" section below, this should now be in a "Non-Requirement" section. IOW you are no longer considering this as a requirement for this iteration of this feature; you've decided against it.

Reply:
fixed